### PR TITLE
Support Diktat 2.0.0

### LINKED ISSUE #1898: Support DiKTat 2.0
DiKTat's maven coordinate had been changed from `org.cqfn.diktat:diktat-rules` to `com.saveourtool.diktat:diktat-rules`, it's 2.0.0 RC state for now, we can support it and drop the support for 1.x versions.

https://github.com/saveourtool/diktat/releases/tag/v2.0.0-rc.8
https://central.sonatype.com/search?q=com.saveourtool.diktat

### PR DESCRIPTION
**Pull request by nulls**
Please **DO NOT FORCE PUSH**. Don't worry about messy history, it's easier to do code review if we can tell what happened after the review, and force pushing breaks that.

Please make sure that your [PR allows edits from maintainers](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/).  Sometimes its faster for us to just fix something than it is to describe how to fix it.

Allow edits from maintainers

After creating the PR, please add a commit that adds a bullet-point under the `[Unreleased]` section of [CHANGES.md](https://github.com/diffplug/spotless/blob/main/CHANGES.md), [plugin-gradle/CHANGES.md](https://github.com/diffplug/spotless/blob/main/plugin-gradle/CHANGES.md), and [plugin-maven/CHANGES.md](https://github.com/diffplug/spotless/blob/main/plugin-maven/CHANGES.md) which includes:

- [ ] a summary of the change
- either
    - [ ] a link to the issue you are resolving (for small changes)
    - [ ] a link to the PR you just created (for big changes likely to have discussion)

If your change only affects a build plugin, and not the lib, then you only need to update the `plugin-foo/CHANGES.md` for that plugin.

If your change affects lib in an end-user-visible way (fixing a bug, updating a version) then you need to update `CHANGES.md` for both the lib and all build plugins.  Users of a build plugin shouldn't have to refer to lib to see changes that affect them.

This makes it easier for the maintainers to quickly release your changes :)

It closes #1898.

### CODE REVIEW

**Comment by Goooler** [comment]
Closes #1898.

File: lib/src/compatDiktat2Dot0Dot0/java/com/diffplug/spotless/glue/diktat/compat/DiktatCompat2Dot0Dot0Adapter.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.glue.diktat.compat;
+
+import java.io.*;
```
Comment (Goooler) [inline lib/src/compatDiktat2Dot0Dot0/java/com/diffplug/spotless/glue/diktat/compat/DiktatCompat2Dot0Dot0Adapter.java:18]: Shouldn't use star imports.

File: lib/src/compatDiktatApi/java/com/diffplug/spotless/glue/diktat/compat/DiktatReporting.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * limitations under the License.
+ */
+package com.diffplug.spotless.glue.diktat.compat;
+
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.ToIntFunction;
+
+final class DiktatReporting {
+	static <T> void reportIfRequired(List<T> errors, ToIntFunction<T> lineGetter, ToIntFunction<T> columnGetter, Function<T, String> detailGetter) {
```
Comment (Goooler) [inline lib/src/compatDiktatApi/java/com/diffplug/spotless/glue/diktat/compat/DiktatReporting.java:23]:
```suggestion
interface DiktatReporting {
	static <T> void reportIfRequired(
		List<T> errors,
		ToIntFunction<T> lineGetter,
		ToIntFunction<T> columnGetter,
		Function<T, String> detailGetter) {
```

File: lib/src/main/java/com/diffplug/spotless/kotlin/DiktatStep.java [hunk 3]
```diff
@@ -24,18 +25,22 @@
...
+	private static final String PACKAGE_RELOCATED_VERSION = "2.0.0";
+
+	private static final String DEFAULT_VERSION = "2.0.0";
 	static final String NAME = "diktat";
-	static final String PACKAGE_DIKTAT = "org.cqfn.diktat";
-	static final String MAVEN_COORDINATE = PACKAGE_DIKTAT + ":diktat-rules:";
+	static final String PACKAGE_PRE_2_0_0 = "org.cqfn.diktat";
+	static final String PACKAGE = "com.saveourtool.diktat";
+	static final String MAVEN_COORDINATE_PRE_2_0_0 = PACKAGE_PRE_2_0_0 + ":diktat-rules:";
+	static final String MAVEN_COORDINATE = PACKAGE + ":diktat-runner:";
```
Comment (Goooler) [inline lib/src/main/java/com/diffplug/spotless/kotlin/DiktatStep.java:43]:
```suggestion
	private static final String MAVEN_COORDINATE_PRE_2_0_0 = "org.cqfn.diktat:diktat-rules:";
	private static final String MAVEN_COORDINATE = "com.saveourtool.diktat:diktat-runner:";
```

File: lib/src/main/java/com/diffplug/spotless/kotlin/DiktatStep.java [hunk 4]
```diff
@@ -24,18 +25,22 @@
...
 	// prevent direct instantiation
 	private DiktatStep() {}
 
 	private static final String MIN_SUPPORTED_VERSION = "1.2.1";
 
-	private static final String DEFAULT_VERSION = "1.2.5";
+	private static final String PACKAGE_RELOCATED_VERSION = "2.0.0";
+
+	private static final String DEFAULT_VERSION = "2.0.0";
 	static final String NAME = "diktat";
```
Comment (Goooler) [inline lib/src/main/java/com/diffplug/spotless/kotlin/DiktatStep.java:39]:
```suggestion
	private static final String NAME = "diktat";
```

File: lib/src/main/java/com/diffplug/spotless/kotlin/DiktatStep.java [hunk 5]
```diff
@@ -68,29 +73,34 @@ static final class State implements Serializable {
...
 
 		State(String versionDiktat, Provisioner provisioner, boolean isScript, @Nullable FileSignature config) throws IOException {
-
 			HashSet<String> pkgSet = new HashSet<>();
-			pkgSet.add(MAVEN_COORDINATE + versionDiktat);
+			if (BadSemver.version(versionDiktat) >= BadSemver.version(PACKAGE_RELOCATED_VERSION)) {
+				pkgSet.add(MAVEN_COORDINATE + versionDiktat);
+			} else {
+				pkgSet.add(MAVEN_COORDINATE_PRE_2_0_0 + versionDiktat);
+			}
```
Comment (Goooler) [inline lib/src/main/java/com/diffplug/spotless/kotlin/DiktatStep.java:88]:
```suggestion
			final String diktatCoordinate;
			if (BadSemver.version(versionDiktat) >= BadSemver.version(PACKAGE_RELOCATED_VERSION)) {
				diktatCoordinate = MAVEN_COORDINATE + versionDiktat;
			} else {
				diktatCoordinate = MAVEN_COORDINATE_PRE_2_0_0 + versionDiktat;
			}
			this.jar = JarState.from(diktatCoordinate, provisioner);
```

**Review by Goooler — APPROVED** [review]

File: lib/src/compatDiktat2Dot0Dot0/java/com/diffplug/spotless/glue/diktat/compat/DiktatCompat2Dot0Dot0Adapter.java (hunk 1 above)
Comment (nulls) [reply lib/src/compatDiktat2Dot0Dot0/java/com/diffplug/spotless/glue/diktat/compat/DiktatCompat2Dot0Dot0Adapter.java]: fixed

**Comment by nulls** [comment]
@nedtwigg could you please review this PR?

**Comment by nedtwigg** [comment]
Great PR, sorry for the long delay! I'll release it shortly :)

**Comment by nedtwigg** [comment]
Published in `plugin-maven 2.42.0` and `plugin-gradle 6.24.0`.

**Pull request merged**